docs/rules/no-unnecessary-polyfills.md

```diff
@@ -0,0 +1,42 @@
+# Enforce the use of built-in methods instead of unnecessary polyfills
+
+This rules helps to use existing methods instead of using extra polyfills. 
+
+## Fail
+
+package.json
+```json
+{
+	"engines": {
+		"node": ">=8"
+	}
+}
+
+```
+```js
+const objectAssign = require('object-assign');
+```
+
+## Pass
+
+package.json
+```json
+{
+	"engines": {
+		"node": "<4.0.0"
+	}
+}
+
+```
+```js
+const assign = require("object-assign"); // passes as Object.assign is not supported
+```
+
+## Options
+
+By default it takes node version from `package.json`. However you can override
+target node version:
+
+```js
+"unicorn/no-unnecessary-polyfills": ["error", {"targetVersion": "6"}]
```

Comment:
I think the option should be called `nodeVersion` and you need to document that it can be a semver range. The example here is also wrong as `6` will only match exactly Node.js `6.0.0`, not `6.0.1`.